addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/CFactory.java

```diff
@@ -93,4 +67,80 @@
         }
         return createComponent(thingUID, haID, configJSON, updateListener, gson);
     }
+
+    protected static class Builder {
+        private ThingUID thingUID;
+        private HaID haID;
+        private String configJSON;
+        private @Nullable ChannelStateUpdateListener updateListener;
+        private Gson gson;
+
+        protected Builder(ThingUID thingUID, HaID haID, String configJSON, Gson gson) {
+            this.thingUID = thingUID;
+            this.haID = haID;
+            this.configJSON = configJSON;
+            this.gson = gson;
+        }
+
+        public Builder withListerner(@Nullable ChannelStateUpdateListener updateListener) {
+            this.updateListener = updateListener;
+            return this;
+        }
+
+        @Nullable
+        public AbstractComponent<?> build() {
```

Comment:
I think at the moment the situation might be a bit confusing for fellow developers.

We are using the factory pattern here. Our factory class has a `create` method, all right.
But in the factory we have a Builder, following the builder pattern. Unfortunately not entirely as expected.

A builder inside of a class is expected to build that class, so here a fellow developer would expect the builder to build factories, which is not quite right of course.

I suggest that you rename "Builder" to "ComponentConfiguration" or similar. Because you are actually not using the builder pattern here, but you are using this object for wrapping multiple configurations, dedicated for components, in one object. This is obvious because you have Getters on this class, which a builder usually does not have.

I also suggest to move the "build" method back to the factory, actually move the content back to the create method to make the factory a real factory again. (Just a suggestion, but it's easier for others if we follow the patterns.)